pkg/agent/apiserver/handlers/ovsflows/handler_test.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ovsflows
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	interfacestoretest "github.com/vmware-tanzu/antrea/pkg/agent/interfacestore/testing"
+	oftest "github.com/vmware-tanzu/antrea/pkg/agent/openflow/testing"
+	"github.com/vmware-tanzu/antrea/pkg/agent/querier"
+	aqtest "github.com/vmware-tanzu/antrea/pkg/agent/querier/testing"
+	networkingv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/networking/v1beta1"
+	ofctltest "github.com/vmware-tanzu/antrea/pkg/ovs/ofctl/testing"
+	queriertest "github.com/vmware-tanzu/antrea/pkg/querier/testing"
+)
+
+var (
+	testFlowKeys    = []string{"flowKey1", "flowKey2"}
+	testDumpResults = []string{"flow1", "flow2"}
+	testResponses   = []Response{{"flow1"}, {"flow2"}}
+)
+
+type testCase struct {
+	test           string
+	name           string
+	namespace      string
+	query          string
+	expectedStatus int
+}
+
+func TestBadRequests(t *testing.T) {
+	badRequests := map[string]string{
```

Comment:
not worth using subtests for these you think?

Reply:
The problem is some cases might not fall into any of Pod, NetworkPolicy, Table flows.
Also the handler implementation validate requests first before calling any Agent funcs. So, might be good to keep requests validation separate too?

Reply:
Agreed. I was just wondering why you didn't use the subtest pattern like you did for other tests such as `TestPodFlows`.